File: backend/Intex2/Data/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace RootkitAuth.API.Data
{
    public class Movie
    {
        [Key]
        public string show_id { get; set; } // varchar in DB

        [Required]
        public string type { get; set; }

        public string title { get; set; }
        public string? director { get; set; }
        public string? cast { get; set; }
        public string? country { get; set; }
        public short release_year { get; set; }  // ? matches SQL smallint
        public string? rating { get; set; }
        public string? duration { get; set; }
        public string? description { get; set; }

        public byte? Action { get; set; }
        public byte? Adventure { get; set; }
        public byte? Anime_Series_International_TV_Shows { get; set; }
        public byte? British_TV_Shows_Docuseries_International_TV_Shows { get; set; }
        public byte? Children { get; set; }
        public byte? Comedies { get; set; }
        public byte? Comedies_Dramas_International_Movies { get; set; }
        public byte? Comedies_International_Movies { get; set; }
        public byte? Comedies_Romantic_Movies { get; set; }
        public byte? Crime_TV_Shows_Docuseries { get; set; }
        public byte? Documentaries { get; set; }
        public byte? Documentaries_International_Movies { get; set; }
        public byte? Docuseries { get; set; }
        public byte? Dramas { get; set; }
        public byte? Dramas_International_Movies { get; set; }
        public byte? Dramas_Romantic_Movies { get; set; }
        public byte? Family_Movies { get; set; }
        public byte? Fantasy { get; set; }
        public byte? Horror_Movies { get; set; }
        public byte? International_Movies_Thrillers { get; set; }
        public byte? International_TV_Shows_Romantic_TV_Shows_TV_Dramas { get; set; }
        public byte? Kids_TV { get; set; }
        public byte? Language_TV_Shows { get; set; }
        public byte? Musicals { get; set; }
        public byte? Nature_TV { get; set; }
        public byte? Reality_TV { get; set; }
        public byte? Spirituality { get; set; }
        public byte? TV_Action { get; set; }
        public byte? TV_Comedies { get; set; }
        public byte? TV_Dramas { get; set; }
        public byte? Talk_Shows_TV_Comedies { get; set; }
        public byte? Thrillers { get; set; }
    }
}

File: frontend/src/types/Movie.ts
export interface Movie {
  show_id: string;
  type: string;
  title: string;
  director: string;
  cast: string;
  country: string;
  release_year: string;
  rating: string;
  duration: string;
  description: string;
  genre: string;
  Action: boolean;
  Adventure: boolean;
  Anime_Series_International_TV_Shows: boolean;
  British_TV_Shows_Docuseries_International_TV_Shows: boolean;
  Children_Family_Movies: boolean;
  Children: boolean;
  Comedies: boolean;
  Comedies_Dramas_International_Movies: boolean;
  Comedies_International_Movies: boolean;
  Comedies_Romantic_Movies: boolean;
  Crime_TV_Shows_Docuseries: boolean;
  Documentaries: boolean;
  Documentaries_International_Movies: boolean;
  Docuseries: boolean;
  Dramas: boolean;
  Dramas_International_Movies: boolean;
  Dramas_Romantic_Movies: boolean;
  Family_Movies: boolean;
  Fantasy: boolean;
  Horror_Movies: boolean;
  International_Movies_Thrillers: boolean;
  International_TV_Shows_Romantic_TV_Shows_TV_Dramas: boolean;
  Kids_TV: boolean;
  Language_TV_Shows: boolean;
  Musicals: boolean;
  Nature_TV: boolean;
  Reality_TV: boolean;
  Spirituality: boolean;
  TV_Action: boolean;
  TV_Comedies: boolean;
  TV_Dramas: boolean;
  Talk_Shows_TV_Comedies: boolean;
  Thrillers: boolean;
}

export const GENRES: (keyof Omit<
  Movie,
  | "show_id"
  | "type"
  | "title"
  | "director"
  | "cast"
  | "country"
  | "release_year"
  | "rating"
  | "duration"
  | "description"
  | "genre"
>)[] = [
  "Action",
  "Adventure",
  "Anime_Series_International_TV_Shows",
  "British_TV_Shows_Docuseries_International_TV_Shows",
  "Children",
  "Comedies",
  "Comedies_Dramas_International_Movies",
  "Comedies_International_Movies",
  "Comedies_Romantic_Movies",
  "Crime_TV_Shows_Docuseries",
  "Documentaries",
  "Documentaries_International_Movies",
  "Docuseries",
  "Dramas",
  "Dramas_International_Movies",
  "Dramas_Romantic_Movies",
  "Family_Movies",
  "Fantasy",
  "Horror_Movies",
  "International_Movies_Thrillers",
  "International_TV_Shows_Romantic_TV_Shows_TV_Dramas",
  "Kids_TV",
  "Language_TV_Shows",
  "Musicals",
  "Nature_TV",
  "Reality_TV",
  "Spirituality",
  "TV_Action",
  "TV_Comedies",
  "TV_Dramas",
  "Talk_Shows_TV_Comedies",
  "Thrillers",
];
